packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/ClientEventService.java

```diff
@@ -171,24 +186,41 @@ private void send(final Object message, final List<SseEmitter> emitterList, fina
 	 * @param channel the channel to send the message on
 	 * @throws IOException if there was an error sending the message
 	 */
-	// TODO: use anonymous queues, currently this wont behave correctly with
-	// multiple hmi-server instances. Issue #2679
 	@RabbitListener(
-			queues = {"${terarium.client-user-event-queue}"},
-			concurrency = "1")
+			bindings =
+					@QueueBinding(
+							value =
+									@Queue(
+											value =
+													"#{T(java.util.UUID).randomUUID().toString().concat('-user-event-queue')}",
+											durable = "false",
+											autoDelete = "true"),
+							exchange = @Exchange(value = CLIENT_USER_EVENT_EXCHANGE, type = "fanout")))
 	void onSendToUserEvent(final Message message, final Channel channel) throws IOException {
 		final JsonNode messageJson = decodeMessage(message, JsonNode.class);
 		if (messageJson == null) {
 			return;
 		}
 
-		final String userId = messageJson.at("/userId").asText();
-		synchronized (userIdToEmitter) {
-			final List<SseEmitter> emitterList = userIdToEmitter.get(userId);
+		synchronized (userIdToEmitters) {
+			final String userId = messageJson.at("/userId").asText();
+			final List<SseEmitter> emitterList = userIdToEmitters.get(userId);
 			if (emitterList != null) {
-				send(messageJson.at("/event"), emitterList, userId);
+				final Set<SseEmitter> emittersToRemove = new HashSet<>();
```

Comment:
Again to be consistent this should be using synchronize (either all should or all should not)

Reply:
fixed